Composer/packages/server/src/services/project.ts

```diff
@@ -452,36 +451,81 @@ export class BotProjectService {
 
       BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Bot files created'));
 
-      const id = await BotProjectService.openProject(newProjRef, user);
-
-      // in the case of PVA, we need to update the eTag and alias used by the import mechanism
-      createFromPva && BotProjectService.setProjectLocationData(id, { alias, eTag });
-
-      const currentProject = await BotProjectService.getProjectById(id, user);
-
-      // inject shared content into every new project.  this comes from assets/shared
-      !createFromPva &&
-        (await AssetService.manager.copyBoilerplate(currentProject.dataDir, currentProject.fileStorage));
-
-      if (currentProject !== undefined) {
-        !createFromPva && (await ejectAndMerge(currentProject, jobId));
-        BackgroundProcessManager.updateProcess(jobId, 202, formatMessage('Initializing bot project'));
-        await currentProject.updateBotInfo(name, description, preserveRoot);
-
-        if (schemaUrl && !createFromPva) {
-          await currentProject.saveSchemaToProject(schemaUrl, locationRef.path);
-        }
+      const botsToProcess: { storageId: string; path: string; name: string }[] = [];
 
-        await currentProject.init();
+      // The outcome of our creation might be > 1 bot! We need to determine how many bots we find in this folder.
+      // is this a single bot?
+      if (await StorageService.checkIsBotFolder(newProjRef.storageId, newProjRef.path, user)) {
+        botsToProcess.push({ ...newProjRef, name: name });
```

Comment:
nit can use shorthand name instead of name:name

Reply:
done